include/clap/version.h

```diff
@@ -20,11 +20,15 @@ typedef struct clap_version {
 }
 #endif
 
-#define CLAP_VERSION_MAJOR ((uint32_t)1)
-#define CLAP_VERSION_MINOR ((uint32_t)1)
-#define CLAP_VERSION_REVISION ((uint32_t)5)
+#define CLAP_VERSION_MAJOR_DIGIT 1
+#define CLAP_VERSION_MINOR_DIGIT 1
+#define CLAP_VERSION_REVISION_DIGIT 5
+#define CLAP_VERSION_MAJOR ((uint32_t)CLAP_VERSION_MAJOR_DIGIT)
+#define CLAP_VERSION_MINOR ((uint32_t)CLAP_VERSION_MINOR_DIGIT)
+#define CLAP_VERSION_REVISION ((uint32_t)CLAP_VERSION_REVISION_DIGIT)
 #define CLAP_VERSION_INIT                                                                          \
    { CLAP_VERSION_MAJOR, CLAP_VERSION_MINOR, CLAP_VERSION_REVISION }
+#define CLAP_VERSION_NUMBER (CLAP_VERSION_MAJOR_DIGIT << 16) + (CLAP_VERSION_MINOR_DIGIT << 8) + CLAP_VERSION_REVISION_DIGIT
```

Comment:
You need `()` around the whole expression here

Reply:
Actually you may also want to force the type to `uint32_t`?

Reply:
You could also use `CLAP_VERSION_MAJOR` in the construction instead of the digit one.

Reply:
One other idea would be to have:
```C
static int clap_version_cmp(const clap_version_t *v1, const clap_version_t *v2);
```

Reply:
The problem is we want a compile time check so a function won’t do it. That’s why I had to knock out the uints since they confused the pre processor. But yea I definitely need parens 

the other option which may be way less disruptive is just define a clap-version-number as 0x010105 by hand and add a failure to the compile test in c++17 with a static assert. Then we don’t have to change folks who parse this 

if folks parse it we should add that as a comment too :)

lemme know which one you prefer and I can change it tomorrow am nyc time

Reply:
A constexpr function would work for C++. Not for C of course

Reply:
Can you skip a `#include` with a constexpr function? That was the original motivation here.

Reply:
Why do you need to skip a `#include` based upon clap version?

Reply:
That was the original discussion which spawned this. Trying to make code which works with draft extensions moving around 

https://github.com/free-audio/clap/discussions/252#discussioncomment-4478704